composer/models/huggingface.py

```diff
@@ -88,23 +87,33 @@ def forward(self, batch):
     def loss(self, outputs, batch):
         return outputs['loss']
 
-    def validate(self, batch):
+    def eval_forward(self, batch, outputs: Optional[Any] = None):
         if self.use_logits:
-            labels = batch.pop('labels')
-            output = self.forward(batch)
+            if outputs:
+                output = outputs
+            else:
+                output = self.forward(batch)
```

Comment:
```suggestion
            output = outputs if outputs else self.forward(batch)
```

Reply:
This can probalby be i the outer loop actually, something like:
```
def eval_forward(..):
    output = outputs if outputs else self.forward(batch)
    if self.use_logits:
        output = output['logits']..
        ...
         return output
     return output